### PR TITLE
Share of deposits

### PR DESCRIPTION
**Pull request by kalkov**
#### Summary

Share of deposits from Daily Active Addresses

**Note**
`activeAddresses` & `activeDeposits` are temporary disabled due to
unconsistent data with Daily Active Addresses from TimescaleDB.

The fields will be bringed back after DAA is refactored to use
Clickhouse.

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase/clickhouse/share_of_deposits/eth_share_of_deposits.ex [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  @spec share_of_deposits(
+          %DateTime{},
+          %DateTime{},
+          String.t()
+        ) :: {:ok, list(share_of_deposits)} | {:error, String.t()}
+  def share_of_deposits(from, to, interval) do
+    {query, args} = share_of_deposits_query(from, to, interval)
+
+    ClickhouseRepo.query_transform(query, args, fn [
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/share_of_deposits/eth_share_of_deposits.ex]: can you put each arg on a new line manually and run the formatter after that. The weird function format will be fixed then

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 2]
```diff
@@ -197,4 +198,36 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
...
+             interval,
+             @one_hour_seconds,
+             50
+           ),
+         {:ok, share_of_deposits} <-
+           ShareOfDeposits.share_of_deposits(contract, from, to, interval) do
+      {:ok, share_of_deposits}
+    else
+      {:error, {:missing_contract, error_msg}} ->
+        {:error, error_msg}
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Should this return error or just `{:ok, []}` :?

File: test/sanbase/clickhouse/share_of_deposits/erc20_share_of_deposits_test.exs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  test "returns share of deposits from daily active addresses",
+       context do
+    with_mock Sanbase.ClickhouseRepo,
+      query: fn _, _ ->
+        {:ok,
+         %{
+           rows: [
+             [from_iso8601_to_unix!("2019-01-01T00:00:00Z"), 200_000, 20_000, 10.0],
```
Comment (IvanIvanoff) [inline test/sanbase/clickhouse/share_of_deposits/erc20_share_of_deposits_test.exs]: `DateTime.to_unix!(context.from)`

File: test/sanbase_web/graphql/clickhouse/share_of_deposits_test.exs [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  import ExUnit.CaptureLog
+  import Sanbase.Factory
+
+  alias Sanbase.Clickhouse.ShareOfDeposits
+
+  setup do
+    user = insert(:staked_user)
+    conn = setup_jwt_auth(build_conn(), user)
+
+    token = insert(:project, %{main_contract_address: "0x123"})
```
Comment (IvanIvanoff) [inline test/sanbase_web/graphql/clickhouse/share_of_deposits_test.exs]: `project`

Comment (kalkov) [reply test/sanbase_web/graphql/clickhouse/share_of_deposits_test.exs]: I want to explicitly differentiate between token and ethereum.

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 2 above)
Comment (kalkov) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Well, as a user I would expect an error and we have same behavior in the other places in the codebase?

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: My issue with this `:error` is that it won't be cached and all calls to this API will make DB calls. Also the the metrics will report this as error response while nothing wrong happened - we just don't have data for that project.

Comment (kalkov) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Let's address in another PR

**Pull request merged**